pyproject.toml

```diff
@@ -101,6 +101,7 @@ docformatter = "==1.7.5"
 mypy = "==1.8.0"
 pylint = "==3.0.3"
 flake8 = "==5.0.4"
+parameterized = "==0.9.0"
```

Comment:
It would be better to amend project dependencies in a separate PR. @danieljanes Do you think we should add `parameterized` to our dev env?

Reply:
@panh99 We need the `parameterized` for the unit tests in this PR.

Reply:
Yeah, I understand. Just talked to Daniel, and I will open a separate PR for adding the dependency. It's generally good to keep each PR focusing on a single topic.